simd/performance_tests/include/Common.hpp

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright Contributors to the Kokkos project
+
+#ifndef KOKKOS_SIMD_PERFTEST_COMMON_HPP
+#define KOKKOS_SIMD_PERFTEST_COMMON_HPP
+
+#include <Kokkos_Core.hpp>
+#include <Kokkos_SIMD.hpp>
+#include <Kokkos_Random.hpp>
+#include <cstdlib>
+#include <benchmark/benchmark.h>
+
+template <typename T, typename ExecSpace>
+using View =
+    Kokkos::View<T, ExecSpace,
+                 Kokkos::MemoryTraits<Kokkos::Aligned | Kokkos::Restrict>>;
+
+// Class wrapping the Arguments we use in the benchmarks.
+// This avoids recreating the Views between each benches.
+template <class T, class ExecSpace>
+struct Args {
+  using view_type = View<T*, ExecSpace>;
+
+  view_type arg1;
+  view_type arg2;
+  view_type arg3;
+
+  explicit Args(std::size_t size)
+      : arg1("arg1", size), arg2("arg2", size), arg3("arg3", size) {
+    Kokkos::Random_XorShift64_Pool<ExecSpace> random_pool(58051);
+
+    Kokkos::fill_random(arg1, random_pool, 1, 100);
+    Kokkos::fill_random(arg2, random_pool, 1, 15);
+    Kokkos::fill_random(arg3, random_pool, 1, 100);
+  }
+};
+
+// Google Benchmark keeps a pointer to the data passed as argument when
+// registering a benchmark.
+// If we don't manually destroy the Views, they will get deleted after
+// Kokkos::finalize() is called, causing a Kokkos::abort.
+
+// Hook that will be called right before Kokkos::finalize(), ensuring that the
+// Views are deleted in time.
+template <class T, class ExecSpace>
+struct FinalizeHook {
+  Args<T, ExecSpace>* args;
+
+  void operator()() { delete args; }
+};
+
+// Class wrapping the arguments, Google Benchmark will have an handle on this
+// class and not on the underlying Args class.
+template <class T, class ExecSpace>
+struct ArgsWrapper {
+  Args<T, ExecSpace>* args;
+
+  explicit ArgsWrapper(std::size_t size) {
+    args = new Args<T, ExecSpace>(size);
+
+    FinalizeHook hook(args);
+    Kokkos::push_finalize_hook(hook);
+  }
+};
```

Comment:
Does it work with a lambda? It would make this part a bit easier to understand
```suggestion
// Class wrapping the arguments, Google Benchmark will have an handle on this
// class and not on the underlying Args class.
template <class T, class ExecSpace>
struct ArgsWrapper {
  Args<T, ExecSpace>* args;

  explicit ArgsWrapper(std::size_t size) {
    args = new Args<T, ExecSpace>(size);

    Kokkos::push_finalize_hook([args]() { delete args; });
  }
};
```

Reply:
I couldn't make it work with a lambda.
Your solution doesn't work because it gives the errors `invalid use of non-static data member` and we can't capture `this` either.

Reply:
If the error comes from capturing this, you can try introducing a variable
```cpp
Kokkos::push_finalize_hook([ptr=args]() { delete ptr; });
```
It seems to compile fine on my end with gcc, icpx and nvcc

Reply:
But anyway, I get a segfault related to the Wrapper on cuda when running the DevicePerfTest, so I think we need to think of an alternative solution to this issue